SingularityBase/src/main/java/com/hubspot/mesos/SingularityDockerInfo.java

```diff
@@ -16,26 +17,41 @@
   private final Optional<SingularityDockerNetworkType> network;
   private final List<SingularityDockerPortMapping> portMappings;
   private final boolean forcePullImage;
-  private final Map<String, String> parameters;
+  private final Optional<Map<String, String>> parameters;
+  private final List<SingularityDockerParameter> dockerParameters;
 
   @JsonCreator
   public SingularityDockerInfo(@JsonProperty("image") String image,
                                @JsonProperty("privileged") boolean privileged,
                                @JsonProperty("network") SingularityDockerNetworkType network,
                                @JsonProperty("portMappings") Optional<List<SingularityDockerPortMapping>> portMappings,
                                @JsonProperty("forcePullImage") Optional<Boolean> forcePullImage,
-                               @JsonProperty("parameters") Optional<Map<String, String>> parameters) {
+                               @JsonProperty("parameters") Optional<Map<String, String>> parameters,
+                               @JsonProperty("dockerParameters") Optional<List<SingularityDockerParameter>> dockerParameters) {
+    if (dockerParameters.isPresent() && !dockerParameters.get().isEmpty() && parameters.isPresent() && !parameters.get().isEmpty()) {
+      throw new IllegalArgumentException("Can only specify one of 'parameters' or 'dockerParameters'");
+    }
```

Comment:
I'm a little hesitant of putting these kinds of exceptions in constructors -- if we somehow had a bogus one saved we'd be throwing every time it was deserialized. What do you think about moving this into `SingularityValidator` instead?